regionServiceClient/lib/cloudregister/registerutils.py

```diff
@@ -547,12 +577,38 @@ def replace_hosts_entry(current_smt, new_smt):
     """Replace the information of the SMT server in /etc/hosts"""
     known_hosts = open(HOSTSFILE_PATH, 'r').readlines()
     new_hosts = ''
-    current_smt_ip = current_smt.get_ip()
+    current_smt_ipv4 = current_smt.get_ipv4()
+    current_smt_ipv6 = current_smt.get_ipv6()
+    smt_ipv6_access = has_ipv6_access(new_smt)
+    new_entry = '%s\t' + new_smt.get_FQDN() + '\t' + new_smt.get_name() + '\n'
     for entry in known_hosts:
-        if current_smt_ip in entry:
-            new_hosts += '%s\t%s\t%s\n' % (new_smt.get_ip(),
-                                           new_smt.get_FQDN(),
-                                           new_smt.get_name())
+        if (
```

Comment:
This is a bit confusing. Consider using a tiered if else block here instead of multiple if's with continues. Seems like it could be simplified because if smt_ipv6_access then get_ipv6() is always used. And vice versa with ipv4.